src/ripple/json/impl/json_value.cpp

```diff
@@ -245,36 +213,35 @@ Value::Value ( double value )
 
 Value::Value ( const char* value )
     : type_ ( stringValue )
-    , allocated_ ( true )
 {
-    value_.string_ = valueAllocator ()->duplicateStringValue ( value );
-}
-
+    if (!value)
+    {
+        value_.string_ = emptyString;
+        return;
+    }
 
-Value::Value ( const char* beginValue,
-               const char* endValue )
-    : type_ ( stringValue )
-    , allocated_ ( true )
-{
-    value_.string_ = valueAllocator ()->duplicateStringValue ( beginValue,
-                     UInt (endValue - beginValue) );
+    allocated_ = true;
+    value_.string_ = dupString(value);
 }
 
-
 Value::Value ( std::string const& value )
     : type_ ( stringValue )
-    , allocated_ ( true )
 {
-    value_.string_ = valueAllocator ()->duplicateStringValue ( value.c_str (),
-                     (unsigned int)value.length () );
+    if (value.empty())
```

Comment:
this could be deferred to the other const char* constructor, we aren't taking advantage of std::string.size() in dupString, so this let's us eliminate some code. 

```
Value::Value ( std::string const& value ) 
    : Value(value.empty() ? nullptr : value.c_str()) 
{}
```

Reply:
My previous code took advantage of `size` but I decided it wasn't worth the effort. I'll fix.